types/culori/test/clamp.test.ts

```diff
@@ -14,3 +14,19 @@ clampChroma(lch1, mode); // $ExpectType Color | undefined
 clampChroma(lch2, mode); // $ExpectType Lch
 clampChroma(lch1, mode, gamutMode); // $ExpectType Color | undefined
 clampChroma(lch2, mode, gamutMode); // $ExpectType Lch
+
+clampGamut(mode)(lch1); // $ExpectType Color | undefined
+clampGamut(mode)(lch2); // $ExpectType Color | undefined
```

Comment:
I've tried to do here this:
```suggestion
clampGamut(mode)(lch2); // $ExpectType Lch | undefined
```

But didn't manage to make it work. Maybe some bug in TS, because the solution I tried works for `clampChroma`.

Reply:
That’s a good catch—probably means the type can be improved. This does seem like it should also work; you’re right. Will update.

Reply:
Was able to improve the types of the functions I added by quite a bit (didn’t touch any preexisting ones). They were _okay_ before, but you were right—there were a couple bugs that were fixed, and they now infer the color type much better.